README.md

```diff
@@ -267,6 +277,45 @@ The shared storage worklet invocation methods (`addModule`, `run`, and `selectUR
 *  The response header will only be honored if the corresponding request included the request header: `Sec-Shared-Storage-Writable: ?1`.
 *  See example usage below.
 
+### Locking for Modifier Methods
+
+All modifier methods (`set`, `append`, `delete`, `clear`), whether invoked from JavaScript or from response headers, accept a `withLockOnResource: <resource>` option. This option instructs the method to acquire a lock on the designated resource before executing.
+
+The locks requested this way are partitioned by the shared storage data origin, and are independent of any locks obtained via `navigator.locks.request` in a Window or Worker context. Note that they share the same scope with the locks obtained via `navigator.locks.request` in the SharedStorageWorklet context.
+
+Unlike `navigator.locks.request` that offers various configuration options, the locks requested this way always use the default settings:
+*  `mode: "exclusive"`: The lock is never shared with other locks.
+*  `steal: false`: The lock will not preempt other locks.
+*  `ifAvailable: false`: If the lock is currently held by others, keep waiting and don't skip.
+
+#### Caveat 1: Unexpected ordering
+
+Modifier methods may block due to the lock, so may not execute in the order they appear in the code.
+
+```js
+// Resolve immediately. Internally, this may block to wait for the lock to be granted.
+sharedStorage.set('key0', 'value1', { withLockOnResource: 'resource0' });
+
+// Resolve immediately. Internally, this will also execute immediately.
+sharedStorage.set('key0', 'value2');
+```
+
+Developers should be mindful of this potential ordering issue.
+
+#### Caveat 2: No atomicity for multiple writes
+
+The `withLockOnResource` option provides locking for individual methods but doesn't guarantee atomicity across multiple methods.
+
+Developers should be mindful of this limitation. To get around, consider one of these options:
+1.  Move the modifier methods inside the worklet. This allows a single lock to protect them.
+2.  Place the data under the same key and modify it within a single method, which is inherently atomic.
+
+### Recommendations for lock usage
+
+If the worklet performs only a single read or write, the lock is unnecessary (both inside and outside the worklet) because each read or write is inherently atomic. Example: [A/B experiment](#simple-example-consistent-ab-experiments-across-sites).
+
+If the worklet performs both read and write on the same key, the lock is likely necessary. Examples: [cross-site reach measurement](#cross-site-reach-measurement), [creative selection by frequency](#creative-selection-by-frequency).
+
```

Comment:
I like the examples you have below of locks inside the worklet. Could we perhaps have a contextual example of why we would want to use a lock outside the worklet? 

(I know you have the example showing that ordering may be unexpected, but perhaps an example that motivates why one would want to use the option `withLockOnResource` in the first place?)

Reply:
I came up with a naive example. I didn't come up with a more meaningful one: This would involve the worklet doing multiple read and writes, but:
- If the worklet only interacts with a single key (like the "reach" example), it's unclear in what scenarios the key also needs modification outside the worklet.
- If the worklet operates on multiple keys, it likely encounters the limitations outlined in [Caveat #2](https://github.com/WICG/shared-storage/blob/yao-web-locks/README.md#caveat-2-no-atomicity-for-multiple-writes), which makes it unsuitable as a good example.

Reply:
Thank you for adding the example!